File: ruby/testcode/benchmark_test_01217.rb
require_relative 'shared'

def match_user_pattern(req)
  pattern = req.param('pattern')
  text = req.param('text')
  re = Regexp.new(pattern)
  matched = re.match(text)
  BenchmarkResponse.ok(matched ? 'match' : 'no match')
end

File: rust/testcode/benchmark_test_01217.rs
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let data = req.body_str();
    let bytes = data.as_bytes();
    let raw_pos: usize = req.param("pos").parse().unwrap_or(0);
    match bytes.get(raw_pos..raw_pos.saturating_add(4)) {
        Some(chunk) => super::shared::BenchmarkResponse::ok(&format!("Chunk: {:?}", chunk)),
        None => super::shared::BenchmarkResponse::bad_request("Position out of range"),
    }
}
